src/main/webapp/itempagescript.js

```diff
@@ -2,84 +2,136 @@
  * Retrieves ItemPageData and forms page using other functions.
  */
 async function loadItemPage() {
-  $(document).ready(function() {
-    $('#navbar').load('navbar.html', function() {
+  $(document).ready(function () {
+    $("#navbar").load("navbar.html", function () {
       initializeNavBarProfileSection();
     });
-    const itemId = getUrlParam('itemId');
+    const itemId = getUrlParam("itemId");
 
-    if (itemId !== '') {
+    if (itemId !== "") {
       fetch(`/itempagedata?itemId=${itemId}`)
-          .then((response) => response.json())
-          .then((itemPageData) => {
-            createSelectedItemCard(itemPageData.item);
-            getItemPageComments(itemPageData.comments);
-          });
+        .then((response) => response.json())
+        .then((itemPageData) => {
+          fetch("/login")
+            .then((response) => response.json())
+            .then((loginResponse) => {
+              createSelectedItemCard(itemPageData.item);
+              if (loginResponse.isUserLoggedIn) {
+                getItemPageComments(itemPageData.comments);
+              } else {
+                getItemPageComments(itemPageData.comments);
```

Comment:
You can also move `getItemPageComments(itemPageData.comments);` out of the if/else since it is called in both. 
```
createSelectedItemCard(itemPageData.item);
getItemPageComments(itemPageData.comments);
if (!loginResponse.isUserLoggedIn) {
  hideCommentBox();
  ...
}
 ```